### PR TITLE
kubeconform to validate rendered yaml for helm chart

### LINKED ISSUE #1914: incorporate helm chart validation in CI
As suggested by @damienpontifex but splitting out from  #1908 PR here.

```

  xtask_check_helm:
    parameters:
      os:
        type: string
    steps:
      - run:
          name: Validate helm manifests
          command: |
            TODAY=$(date +"%Y-%m-%d")
            CURRENT_KUBE_VERSIONS=$(curl -L https://raw.githubusercontent.com/kubernetes/website/main/data/releases/schedule.yaml \
              | yq -o json '.' \
              | jq --raw-output '.schedules[] | select((now | strftime("%Y-%m-%dT00:00:00Z")) as $date | .releaseDate < $date and .endOfLifeDate > $date) | .previousPatches[0].release')
            TEMPLATE_DIR=$(mktemp -d)
            echo "${CURRENT_KUBE_VERSIONS}" | while read kube_version; do
              helm template --kube-version "${kube_version}" router helm/chart/router --set autoscaling.enabled=true > "${TEMPLATE_DIR}/router-${kube_version}.yaml"
              kubeval "${TEMPLATE_DIR}/router-${kube_version}.yaml" \
                --kubernetes-version "${kube_version}" \
                --strict \
                --schema-location https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master
            done
  xtask_check_compliance:
    parameters:
      os:
	@@ -284,6 +308,22 @@ jobs:
            - linux_install_baseline
            - xtask_lint:
                os: linux
  check_helm:
    environment:
      <<: *common_job_environment
    parameters:
      platform:
        type: executor
    executor: << parameters.platform >>
    steps:
      - checkout
      - when:
          condition:
            equal: [*rust_linux_executor, << parameters.platform >>]
          steps:
            - linux_install_baseline
            - xtask_check_helm:
                os: linux
  check_compliance:
    environment:
      <<: *common_job_environment
	@@ -516,6 +556,11 @@ workflows:
            parameters:
              platform: [rust_linux]

      - check_helm:
          matrix:
            parameters:
              platform: [rust_linux]

      # check-compliance is disabled for now because it is flaky
      # `cargo about` seems to be non-deterministic and sometimes
      # produces a different licenses.html file even though dependencies haven’t changed.
```

### PR DESCRIPTION
**Pull request by damienpontifex**
- Validate yaml for chart against currently supported kubernetes versions
- Originally proposed in #1908
- Would also help avoid issues like the fix required by #2788
- Resolves #1914

**Checklist**

Complete the checklist (and note appropriate exceptions) before a final PR is raised.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]. It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]. Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]. Tick whichever testing boxes are applicable. If you are adding Manual Tests:
    - please document the manual testing (extensively) in the Exceptions.
    - please raise a separate issue to automate the test and label it (or ask for it to be labeled) as `manual test`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -349,6 +349,30 @@ commands:
...
+            TEMPLATE_DIR=$(mktemp -d)
+            echo "${CURRENT_KUBE_VERSIONS}" | while read kube_version; do
+              helm template --kube-version "${kube_version}" router helm/chart/router --set autoscaling.enabled=true > "${TEMPLATE_DIR}/router-${kube_version}.yaml"
+
+              kubeval "${TEMPLATE_DIR}/router-${kube_version}.yaml" \
+                --kubernetes-version "${kube_version}" \
+                --strict \
+                --schema-location https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master
+            done
+
```
Comment (garypen) [inline .circleci/config.yml:375]: [kubeval](https://github.com/instrumenta/kubeval) is a dead project. Can we use [kubeconform](https://github.com/yannh/kubeconform) (or something still active)?

**Review by garypen — COMMENTED** [review]
This is a very welcome PR. #2788 had motivated me to start thinking about this again and it's great to see you've already started to address it.

Let me know what you think about kubeval and I'll do a full review when I hear back from you about that.

File: .circleci/config.yml (hunk 1 above)
Comment (damienpontifex) [reply .circleci/config.yml:375]: We can and should. I copied this from our internal setup, so hadn't considered different tools.

Quick look and kubeconform looks good. I'll update tomorrow, my time, and push changes back for review once done. Equally happy for author contributions if you get to it before I do. Not going to insult me :)

Comment (damienpontifex) [reply .circleci/config.yml]: Updated to use kubeconform

**Comment by garypen** [comment]
@damienpontifex I've taken your bits and munged them around a little to create #2835 . Would you be happy to close this in favour of that other PR? I've tested it with our circleci integration and it is working fine.

**Comment by damienpontifex** [comment]
@garypen very happy to close. Happy for the outcome and not wedded to any single PR.
I'll close here. Thanks

**Pull request closed without merging**